app/models/consent_form.rb

```diff
@@ -161,6 +165,19 @@ def form_steps_for_health_questions
     health_answers.map.with_index { |_args, idx| :"health_#{idx + 1}" }
   end
 
+  def each_health_answer
+    return if health_answers.empty?
+    return to_enum(:each_health_answer) unless block_given?
+
+    ha = health_answers.first
+    loop do
+      yield ha
+      next_health_answer_index = ha.next_health_answer_index
+      break unless next_health_answer_index
+      ha = health_answers[next_health_answer_index]
```

Comment:
I prefer an explicit variable name here, `ha` looks funny. :drum: 

Non-blocking suggestion:

```suggestion
    health_answer = health_answers.first
    loop do
      yield health_answer
      next_health_answer_index = health_answer.next_health_answer_index
      break unless next_health_answer_index
      health_answer = health_answers[next_health_answer_index]
```

Reply:
Also could this become an infinite loop?

Reply:
> Also could this become an infinite loop?

Ya, it could, in the same way a recursive function could infinitely recurse. We could implement something to prevent this, e.g.:

1. run some kind of pre-check, probably when the health answers are created, to check they don't do this
2. track which health answer has already been seen in this loop and bail out if a duplicate is found

These health answers are pretty static ... so the likelihood of this happening seems low, but it's true the impact could be quite high if it does happen. 🤔